src/components/ComboBox/ComboBox.tsx

```diff
@@ -0,0 +1,236 @@
+import { useClickOutside } from '@mantine/hooks';
+import * as Popover from '@radix-ui/react-popover';
+import classnames from 'classnames';
+import { Command, CommandGroup, CommandInput, CommandItem, CommandList } from 'cmdk';
+import React from 'react';
+import * as iconComponents from '../../assets/formatted';
+import { getCommonProps } from '../../utils';
+import { getScssVar } from '../../utils/scssUtils';
+
+export interface ComboBoxProps {
+  /**
+   * List of options to be displayed in the ComboBox.
+   */
+  options: {
+    label?: string;
+    value: string;
```

Comment:
We may want to revisit this in the future to be able to support passing in react nodes for use cases of text and icons.

Reply:
Not a blocker